### PR TITLE
 Add prometheus metrics.

### PR DESCRIPTION
**Pull request by qiuyesuifeng**

### CODE REVIEW

File: server/api/router.go [hunk 1]
```diff
@@ -35,6 +36,7 @@ func createRouter(svr *server.Server) *mux.Router {
 	router.Handle("/api/v1/events", newEventsHandler(svr, rd)).Methods("GET")
 	router.Handle("/api/v1/feed", newFeedHandler(svr, rd)).Methods("GET")
 	router.Handle("/api/v1/history/operators", newHistoryOperatorHandler(svr, rd)).Methods("GET")
+	router.Handle("/api/v1/metrics", prometheus.Handler()).Methods("GET")
```
Comment (qiuyesuifeng) [inline server/api/router.go]: `/metrics` is the default http path in github.com repo example.
Do you think it is better?

File: server/util.go [hunk 2]
```diff
@@ -108,3 +109,25 @@ func (t *slowLogTxn) Commit() (*clientv3.TxnResponse, error) {
 
 	return resp, errors.Trace(err)
 }
+
+// convertName converts variable name to a linux type name.
+// Like `AbcDef -> abc_def`.
+func convertName(str string) string {
+	strBytes := []byte(str)
```
Comment (qiuyesuifeng) [inline server/util.go]: Seems no difference, if we use string, we have to convert travel item rune to []byte.

Comment (siddontang) [inline server/util.go]: why travel rune? you can use 
```
for i:=0; i< len(str); i++ {
 str[i]
}
```

File: server/api/router.go (hunk 1 above)
Comment (siddontang) [inline server/api/router.go]: what's the repo example?

File: server/conn.go [hunk 3]
```diff
@@ -81,18 +81,20 @@ func (c *conn) run() {
 			return
 		}
 
+		start := time.Now()
 		request := msg.GetPdReq()
-		stats.Increment("handle_request")
-		ts := stats.NewTiming()
+		label := convertName(request.GetCmdType().String())
```
Comment (shenli) [inline server/conn.go]: convertName will be called many times. Why not use a dictionary?

Comment (siddontang) [inline server/conn.go]: +1

Comment (qiuyesuifeng) [inline server/conn.go]: Sounds ok.

File: server/util.go [hunk 4]
```diff
@@ -111,23 +110,26 @@ func (t *slowLogTxn) Commit() (*clientv3.TxnResponse, error) {
 }
 
 // convertName converts variable name to a linux type name.
-// Like `AbcDef -> abc_def`.
+// Like `AbcDef -> abc_def`, notice that we must make sure the return name is less than 64 bytes.
 func convertName(str string) string {
```
Comment (siddontang) [inline server/util.go]: any test? 
especially for str len > 64.

**Comment by qiuyesuifeng** [comment]
@shenli  @siddontang @disksing  PTAL

**Comment by shenli** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**